### PR TITLE
feat(sn_node): extra criteria when relocate elder related node

### PR DESCRIPTION
**Pull request by maqi**
1, not to relocate elder to self section, 
2, not counter join_as_relocated node as elder candidate immediately

### CODE REVIEW

File: sn_node/src/node/messaging/membership.rs [hunk 1]
```diff
@@ -222,7 +223,17 @@ impl MyNode {
             cmds.extend(self.relocate_peers(churn_id, excluded_from_relocation)?);
         }
 
-        cmds.extend(self.trigger_dkg()?);
+        // In case the newly joined node is a relocated node, and we have enough elders,
+        // we don't try to promote it even it is qualified.
+        let full_of_elders = self.network_knowledge.elders().len() == elder_count();
+        let all_join_as_relocated = joining_nodes
```
Comment (joshuef) [inline sn_node/src/node/messaging/membership.rs:229]:
```suggestion
        let everything_joining_was_relocated = joining_nodes
```

`joined` ? the check is that everything that's joined us has relocated or not, right?

So, maybe for clarity: `we_should_find_new_elders = !(we_have_enough _elders && everything_joining_was_relocated)` ?

File: sn_node/src/node/relocation.rs [hunk 2]
```diff
@@ -82,7 +82,11 @@ pub(super) fn find_nodes_to_relocate(
     let mut relocating_nodes = vec![];
     for node_state in candidates {
         if node_state.age() == max_age {
-            let dst = dst(&node_state.name(), churn_id);
+            // In case the candidate is an elder, it shall only got relocated to other sections.
```
Comment (joshuef) [inline sn_node/src/node/relocation.rs:85]: Can we add `why` in there, too?

**Review by RolandSherwin — APPROVED** [review]

**Comment by RolandSherwin** [comment]
> Build failed:
> 
>     * [ci](https://github.com/maidsafe/safe_network/actions/runs/3936883956/jobs/6734322471)
Failure will be fixed by #1987, re-trying

**Pull request merged**